packages/cbioportal-utils/src/civic/CivicUtils.ts

```diff
@@ -88,67 +53,58 @@ export function getCivicVariants(
     mutationSpecs?: Array<MutationSpec>
 ): Promise<ICivicVariantIndex> {
     let civicVariants: ICivicVariantIndex = {};
-    let promises: Array<Promise<void>> = [];
-
     if (mutationSpecs) {
-        let calledVariants: Set<number> = new Set([]);
-        for (let mutation of mutationSpecs) {
-            let geneSymbol = mutation.gene.hugoGeneSymbol;
-            let geneEntry = civicGenes[geneSymbol];
-            let proteinChanges = [mutation.proteinChange];
+        const geneToProteinChangeSet: {
+            [geneSymbol: string]: Set<String>;
+        } = mutationSpecs.reduce((acc, mutation) => {
+            const geneSymbol = mutation.gene.hugoGeneSymbol;
             // Match any other variants after splitting the name on + or /
-            let split = mutation.proteinChange.split(/[+\/]/);
-            proteinChanges.push(split[0]);
-            for (let proteinChange of proteinChanges) {
-                if (geneEntry && geneEntry.variants[proteinChange]) {
-                    if (
-                        !calledVariants.has(geneEntry.variants[proteinChange])
-                    ) {
-                        //Avoid calling the same variant
-                        calledVariants.add(geneEntry.variants[proteinChange]);
-                        promises.push(
-                            lookupCivicVariantAndAddToMap(
-                                civicVariants,
-                                geneEntry.variants[proteinChange],
-                                proteinChange,
-                                geneSymbol,
-                                geneEntry.id
-                            )
-                        );
+            const splitProteinChanges = mutation.proteinChange.split(/[+\/]/);
+            if (!acc[geneSymbol]) {
+                acc[geneSymbol] = new Set(splitProteinChanges);
+            } else {
+                for (const splitProteinChange of splitProteinChanges) {
+                    acc[geneSymbol].add(splitProteinChange);
+                }
+            }
+            return acc;
+        }, {} as { [geneSymbol: string]: Set<String> });
+
+        for (const geneSymbol in geneToProteinChangeSet) {
+            if (geneSymbol in civicGenes) {
+                const proteinChangeSet = geneToProteinChangeSet[geneSymbol];
+                const geneVariants = civicGenes[geneSymbol].variants;
+                for (const variantName in geneVariants) {
+                    if (proteinChangeSet.has(variantName)) {
```

Comment:
again, if proteinChangeSet is a subjset of geneVarits, then you could just iterate through proteinChangeSet and make this code clearer.

Reply:
proteinChangeSet is not a subset of geneVarits, they both contain some variants that are not in the other set. We can do something like find shared variants from the two sets and iterate from that list, but it doesn't make the logic easier, so maybe we can keep this